src/OrchardCore.Modules/OrchardCore.AutoSetup/AutoSetupMiddleware.cs

```diff
@@ -0,0 +1,212 @@
+using System;
+using OrchardCore.AutoSetup.Options;
+using OrchardCore.Environment.Shell.Models;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Setup.Services;
+using OrchardCore.Abstractions.Setup;
+
+namespace OrchardCore.AutoSetup
+{
+    /// <summary>
+    /// The auto setup middleware.
+    /// </summary>
+    public class AutoSetupMiddleware
+    {
+        /// <summary>
+        /// The next middleware in the execution pipeline.
+        /// </summary>
+        private readonly RequestDelegate _next;
+
+        /// <summary>
+        /// The Shell settings manager.
+        /// </summary>
+        private readonly IShellSettingsManager _shellSettingsManager;
+
+        /// <summary>
+        /// The auto-setup options.
+        /// </summary>
+        private readonly IOptions<AutoSetupOptions> _setupOptions;
+
+        private readonly IShellHost _shellHost;
+
+        /// <summary>
+        /// The _logger.
+        /// </summary>
+        private readonly ILogger<AutoSetupMiddleware> _logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AutoSetupMiddleware"/> class.
+        /// </summary>
+        /// <param name="next"> The next middleware in the execution pipeline. </param>
+        /// <param name="shellSettingsManager">The Shell settings manager</param>
+        /// <param name="setupOptions"> The auto-setup Options. </param>
+        /// <param name="shellHost"></param>
+        /// <param name="logger"> The logger. </param>
+        public AutoSetupMiddleware(
+            RequestDelegate next,
+            IShellSettingsManager shellSettingsManager,
+            IOptions<AutoSetupOptions> setupOptions,
+            IShellHost shellHost,
+            ILogger<AutoSetupMiddleware> logger)
+        {
+            _logger = logger;
```

Comment:
Maybe put the logger at the end to keep the same order

Reply:
fixed